internal/services/appservice/helpers/web_app_slot_schema.go

```diff
@@ -518,3 +781,262 @@ func FlattenSiteConfigLinuxWebAppSlot(appSiteSlotConfig *web.SiteConfig, healthC
+	if len(siteConfig) == 0 {
+		return nil, nil, nil
+	}
+
+	expanded := &web.SiteConfig{}
+	if existing != nil {
+		expanded = existing
+	}
+
+	currentStack := ""
+
+	winSlotSiteConfig := siteConfig[0]
+
+	if metadata.ResourceData.HasChange("site_config.0.always_on") {
+		expanded.AlwaysOn = utils.Bool(winSlotSiteConfig.AlwaysOn)
+	}
+
+	if metadata.ResourceData.HasChange("site_config.0.api_management_api_id") {
+		expanded.APIManagementConfig = &web.APIManagementConfig{
+			ID: utils.String(winSlotSiteConfig.ApiManagementConfigId),
+		}
+	}
+
+	if metadata.ResourceData.HasChange("site_config.0.api_definition_url") {
+		expanded.APIDefinition = &web.APIDefinitionInfo{
+			URL: utils.String(winSlotSiteConfig.ApiDefinition),
+		}
+	}
+
+	if metadata.ResourceData.HasChange("site_config.0.app_command_line") {
+		expanded.AppCommandLine = utils.String(winSlotSiteConfig.AppCommandLine)
+	}
+
+	if metadata.ResourceData.HasChange("site_config.0.application_stack") {
+		winAppStack := winSlotSiteConfig.ApplicationStack[0]
+		expanded.NetFrameworkVersion = utils.String(winAppStack.NetFrameworkVersion)
+		expanded.PhpVersion = utils.String(winAppStack.PhpVersion)
+		expanded.NodeVersion = utils.String(winAppStack.NodeVersion)
+		expanded.PythonVersion = utils.String(winAppStack.PythonVersion)
+		expanded.JavaVersion = utils.String(winAppStack.JavaVersion)
+		expanded.JavaContainer = utils.String(winAppStack.JavaContainer)
+		expanded.JavaContainerVersion = utils.String(winAppStack.JavaContainerVersion)
+		if winAppStack.DockerContainerName != "" {
+			expanded.WindowsFxVersion = utils.String(fmt.Sprintf("DOCKER|%s/%s:%s", winAppStack.DockerContainerRegistry, winAppStack.DockerContainerName, winAppStack.DockerContainerTag))
+		}
+		currentStack = winAppStack.CurrentStack
+	}
+
+	if metadata.ResourceData.HasChange("site_config.0.virtual_application") {
+		expanded.VirtualApplications = expandVirtualApplicationsForUpdate(winSlotSiteConfig.VirtualApplications)
+	} else {
+		expanded.VirtualApplications = expandVirtualApplications(winSlotSiteConfig.VirtualApplications)
+	}
+
+	if metadata.ResourceData.HasChange("site_config.0.container_registry_use_managed_identity") {
+		expanded.AcrUseManagedIdentityCreds = utils.Bool(winSlotSiteConfig.UseManagedIdentityACR)
+	}
+
+	if metadata.ResourceData.HasChange("site_config.0.container_registry_managed_identity_client_id") {
+		expanded.AcrUserManagedIdentityID = utils.String(winSlotSiteConfig.ContainerRegistryUserMSI)
+	}
+
+	if metadata.ResourceData.HasChange("site_config.0.default_documents") {
+		expanded.DefaultDocuments = &winSlotSiteConfig.DefaultDocuments
+	}
+
+	if metadata.ResourceData.HasChange("site_config.0.http2_enabled") {
+		expanded.HTTP20Enabled = utils.Bool(winSlotSiteConfig.Http2Enabled)
+	}
+
+	if metadata.ResourceData.HasChange("site_config.0.ip_restriction") {
+		ipRestrictions, err := ExpandIpRestrictions(winSlotSiteConfig.IpRestriction)
+		if err != nil {
+			return nil, nil, err
```

Comment:
Since this `ExpandIpRestrictions` is used for a couple attributes, what're your thoughts on expanding on the error message so its' immediately apparent where we went wrong? 

```suggestion
			return nil, nil, fmt.Errorf("expanding `site_config.0.ip_restriction`: %s", err
```

Reply:
It's a good shout, I don't think it should hold this resource up, but I'll add it to my todo list to cover before these resources exit beta 👍